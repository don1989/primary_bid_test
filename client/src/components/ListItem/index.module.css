.root {
	display: flex;
	border: 2px solid rgb(108, 99, 255);
	max-width: 700px;
	margin: 1px;
}

.image {
	width: 50px;
	height: 50px;
}

.item {
	text-align: left;
	width: 100%;
}

.short-url {
	font-weight: bold;
}

.long-url {
	color: green;
}
