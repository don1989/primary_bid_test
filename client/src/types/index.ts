export interface IUrlPair {
	longUrl: string;
	shortUrl: string;
}

export interface IPostRequestBody {
	longUrl: string;
}

export interface IPostResponseBody {
	longUrl: string;
	shortUrl: string;
}

export interface IGetResponseBody {
	urls: IUrlPair[];
}
