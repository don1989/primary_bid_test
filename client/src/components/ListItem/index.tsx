import styles from './index.module.css';
import logo from '../../logo.svg';

import { IUrlPair } from '../../types';

interface Props {
	urlPair: IUrlPair;
}

const ListItem = ({ urlPair }: Props): JSX.Element | null => {
	if (!urlPair.longUrl || !urlPair.shortUrl) {
		return null;
	}
	return (
		<div className={styles.root}>
			<img className={styles.image} src={logo} alt="logo" />
			<div className={styles.item}>
				<div className={styles['short-url']}>
					<a
						href={urlPair.shortUrl}
						target="_blank"
						rel="noreferrer"
					>{`${urlPair.shortUrl}`}</a>
				</div>
				<div className={styles['long-url']}>
					<a
						href={urlPair.longUrl}
						target="_blank"
						rel="noreferrer"
					>{`${urlPair.longUrl}`}</a>
				</div>
			</div>
		</div>
	);
};

export default ListItem;
