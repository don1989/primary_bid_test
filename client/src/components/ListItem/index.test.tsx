import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import ListItem from './';
import { IUrlPair } from '../../types';

describe('ListItem', () => {
	test('renders a list item', () => {
		const urlPair: IUrlPair = {
			longUrl:
				'https://testing-library.com/docs/react-testing-library/api/',
			shortUrl: 'https://pbid.io/hsda9d33',
		};

		const { asFragment, getByText } = render(
			<ListItem urlPair={urlPair} />
		);
		expect(getByText(urlPair.longUrl)).toBeInTheDocument();
		expect(getByText(urlPair.shortUrl)).toBeInTheDocument();
		expect(asFragment()).toMatchSnapshot();
	});

	test('returns null for invalid list item', () => {
		const urlPair: IUrlPair = {
			longUrl: '',
			shortUrl: 'https://pbid.io/hsda9d33',
		};

		const { asFragment } = render(<ListItem urlPair={urlPair} />);
		expect(asFragment()).toMatchSnapshot();
	});
});
