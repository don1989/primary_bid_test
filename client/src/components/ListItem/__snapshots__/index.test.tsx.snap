// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem renders a list item 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <img
      alt="logo"
      class="image"
      src="logo.svg"
    />
    <div
      class="item"
    >
      <div
        class="short-url"
      >
        <a
          href="https://pb.io/hsda9d33"
          rel="noreferrer"
          target="_blank"
        >
          https://pb.io/hsda9d33
        </a>
      </div>
      <div
        class="long-url"
      >
        <a
          href="https://testing-library.com/docs/react-testing-library/api/"
          rel="noreferrer"
          target="_blank"
        >
          https://testing-library.com/docs/react-testing-library/api/
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ListItem returns null for invalid list item 1`] = `<DocumentFragment />`;
